oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -71,7 +71,7 @@
   private static final Logger LOGGER = Logger.getLogger(ComputeEngineCredentials.class.getName());
 
   // Note: the explicit IP address is used to avoid name server resolution issues.
-  static final String DEFAULT_METADATA_SERVER_URL = "http://169.254.169.254";
+  static final String DEFAULT_METADATA_SERVER_URL = "http://metadata.google.internal";
```

Comment:
Please use `metadata.google.internal.` with the `.` at the end.  We need that to solve a nasty problem in docker related to that problem.

Reply:
We had this issue [#4269](https://github.com/googleapis/google-cloud-java/issues/4269) and its fixed with http://metadata.google.internal for GKE and GCE both.